runtime/src/snapshot_utils.rs

```diff
@@ -958,6 +974,96 @@ where
     Ok((bank, timings))
 }
 
+/// Rebuild bank from snapshot archives.  This function searches `snapshot_archives_dir` for the
+/// highest full snapshot and highest corresponding incremental snapshot, then rebuilds the bank.
+#[allow(clippy::too_many_arguments)]
+pub fn bank_from_latest_snapshot_archives(
+    snapshots_dir: impl AsRef<Path>,
+    snapshot_archives_dir: impl AsRef<Path>,
+    account_paths: &[PathBuf],
+    frozen_account_pubkeys: &[Pubkey],
+    genesis_config: &GenesisConfig,
+    debug_keys: Option<Arc<HashSet<Pubkey>>>,
+    additional_builtins: Option<&Builtins>,
+    account_secondary_indexes: AccountSecondaryIndexes,
+    accounts_db_caching_enabled: bool,
+    limit_load_slot_count_from_snapshot: Option<usize>,
+    shrink_ratio: AccountShrinkThreshold,
+    test_hash_calculation: bool,
+    verify_index: bool,
+) -> Result<(Bank, BankFromArchiveTimings)> {
+    let full_snapshot_archive_info = get_highest_full_snapshot_archive_info(&snapshot_archives_dir)
+        .ok_or(SnapshotError::NoSnapshotArchives)?;
+
+    let incremental_snapshot_archive_info = get_highest_incremental_snapshot_archive_info(
+        &snapshot_archives_dir,
+        *full_snapshot_archive_info.slot(),
+    );
+
+    info!(
+        "Loading bank from full snapshot: {}, and incremental snapshot: {:?}",
+        full_snapshot_archive_info.path().display(),
+        incremental_snapshot_archive_info
+            .as_ref()
+            .map(
+                |incremental_snapshot_archive_info| incremental_snapshot_archive_info
+                    .path()
+                    .display()
+            )
+    );
+
+    let (bank, timings) = bank_from_snapshot_archives(
```

Comment:
heh just realized the real hashing/verification of the account state happens in here, when we call down to

`verify_snapshot_bank() -> verify_bank_hash() -> verify_bank_hash_and_lamports() -> calculate_accounts_hash_helper_with_verify()`, and we verify the 
`calculated_hash` of the accounts state against the expected serialized `bank_hashes` in the snapshot